### DISCUSSION TITLE
Issue with @Reducer Macro Showing "Unknown Attribute"

### DISCUSSION THREAD

**Discussion by imhans** [pointfreeco/swift-composable-architecture#2773] [Q&A]
I'm encountering an issue with the TCA framework where the @Reducer macro is being flagged as an "Unknown Attribute" in my code. I'm currently using TCA version 1.4.2 on Ventura 13.2.1 and the module is successfully recognized. I'd like to discuss this problem and seek assistance in resolving it. Any insights or solutions would be greatly appreciated. Thank you!

**Comment by imhans** [comment]
Also, I have tried cleaning build folder and deleting derived folder.

**Reply by mbrandonw** [reply]
Hi @imhans, what version of Xcode are you using? And can you provide a minimal project that demonstrates the problem you are having?

**Comment by andrewayres** [comment]
Hi @mbrandonw - I encountered the same issue using XCode 14.2 (with Swift 5.7.2) on macOS Monterey 12.7.1. Adding TCA as a package (via the main URL) added Branch:main and resulted in TCA v.1.7.3

I was following https://pointfreeco.github.io/swift-composable-architecture/main/tutorials/composablearchitecture/01-01-yourfirstfeature

The 'unknown attribute' error goes away when I change

```
@Reducer
struct CounterFeature {
```

to

`struct CounterFeature: Reducer {`
(a usage I picked up from https://www.pointfree.co/episodes/ep247-tour-of-the-composable-architecture-1-0-correctness)

I also got an 'unknown attribute' error using @ObservableState.

Can I still use @Reducer, @ObservableState, etc., as per the above tutorial with XCode 14.2 and Swift 5.7.2? I want to use your new work on observation, if possible.

**Reply by mbrandonw** [reply]
Hi @andrewayres, Swift macros only work on Xcode 15+, and I believe in just a few months time apps will not be allowed to be submitted to the App Store built with Xcode <15. So I think eventually you will just need to get on Xcode 15 somehow.

**Reply by andrewayres** [reply]
thanks @mbrandonw - that makes it clear for me, it's time to upgrade
